base_url/models/model.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+from openerp import models, api, fields, _
+import logging
+from openerp.exceptions import Warning as UserError
+_logger = logging.getLogger(__name__)
+try:
+    from slugify import slugify
+except ImportError:
+    _logger.debug('Cannot `import slugify`.')
+
+
+class UrlUrl(models.Model):
+
+    _name = "url.url"
+
+    url_key = fields.Char(string="Url Id", store=True)
+    model_id = fields.Reference(selection='_reference_models',
+                                help="The id of product or category.",
+                                readonly=True,
+                                )
+    redirect = fields.Boolean('Redirect')
+
+    _sql_constraints = [("urlurl unique key",
+                         'unique(url_key)',
+                         'Already exists in database')]
+
+    @api.model
+    def _reference_models(self):
+        return []
+
+    @api.multi
+    def _get_object(self, urls):
+        """
+        :return: return object attach to the url
+        """
+        object = self.search([("url_key", "=", urls)]).model_id
+        return object
+
+
+class AbstractUrl(models.AbstractModel):
+    _name = "abstract.url"
+
+    url_key = fields.Char(compute="_compute_url",
+                          inverse="_inverse_set_url", string="Url key")
+    redirect_url_key_ids = fields.One2many(compute="_compute_redirect_url",
+                                           comodel_name="url.url")
+
+    def _prepare_url(self, name=None):
+        url_key = "prepare"
+        if name is None:
```

Comment:
inverser la condition permet de mettre : 
`if name:`  au lieur de  `if name is None:`